pkg/ccl/changefeedccl/helpers_test.go

```diff
@@ -683,6 +585,248 @@ func closeFeed(t testing.TB, f cdctest.TestFeed) {
+		sinkWeights = map[string]int{}
+		for _, sinkType := range options.allowedSinkTypes {
+			sinkWeights[sinkType] = 1
+		}
+	}
+	if options.disabledSinkTypes != nil {
+		for _, sinkType := range options.disabledSinkTypes {
+			sinkWeights[sinkType] = 0
+		}
+	}
+	weightTotal := 0
+	for _, weight := range sinkWeights {
+		weightTotal += weight
+	}
+	p := rand.Float32() * float32(weightTotal)
+	var sum float32 = 0
+	for sink, weight := range sinkWeights {
+		sum += float32(weight)
+		if p <= sum {
+			return sink
+		}
+	}
+	return "kafka" // unreachable
+}
+
+// addCloudStorageOptions adds the options necessary to enable a server to run a
+// cloudstorage changefeed on it
+func addCloudStorageOptions(t *testing.T, options *feedTestOptions) (cleanup func()) {
+	dir, dirCleanupFn := testutils.TempDir(t)
+	options.externalIODir = dir
+	oldKnobsFn := options.knobsFn
+	options.knobsFn = func(knobs *base.TestingKnobs) {
+		if oldKnobsFn != nil {
+			oldKnobsFn(knobs)
+		}
+		blobClientFactory := blobs.NewLocalOnlyBlobClientFactory(options.externalIODir)
+		if serverKnobs, ok := knobs.Server.(*server.TestingKnobs); ok {
+			serverKnobs.BlobClientFactory = blobClientFactory
+		} else {
+			knobs.Server = &server.TestingKnobs{
+				BlobClientFactory: blobClientFactory,
+			}
+		}
+	}
+	return dirCleanupFn
+}
+
+func makeFeedFactory(
+	t *testing.T,
+	sinkType string,
+	s serverutils.TestTenantInterface,
+	db *gosql.DB,
+	testOpts ...feedTestOption,
+) (factory cdctest.TestFeedFactory, sinkCleanup func()) {
+	options := makeOptions(testOpts...)
+	return makeFeedFactoryWithOptions(t, sinkType, s, db, options)
+}
+
+func makeFeedFactoryWithOptions(
+	t *testing.T,
+	sinkType string,
+	s serverutils.TestTenantInterface,
+	db *gosql.DB,
+	options feedTestOptions,
+) (factory cdctest.TestFeedFactory, sinkCleanup func()) {
+	t.Logf("making %s feed factory", sinkType)
+	switch sinkType {
+	case "kafka":
+		f := makeKafkaFeedFactory(s, db)
+		return f, func() {}
+	case "cloudstorage":
+		if options.externalIODir == "" {
+			t.Fatalf("expected externalIODir option to be set")
+		}
+		f := makeCloudFeedFactory(s, db, options.externalIODir)
+		return f, func() {}
+	case "enterprise":
+		sink, cleanup := sqlutils.PGUrl(t, s.SQLAddr(), t.Name(), url.User(username.RootUser))
+		f := makeTableFeedFactory(s, db, sink)
+		return f, cleanup
+	case "webhook":
+		f := makeWebhookFeedFactory(s, db)
+		return f, func() {}
+	case "pubsub":
+		f := makePubsubFeedFactory(s, db)
+		return f, func() {}
+	case "sinkless":
+		sink, cleanup := sqlutils.PGUrl(t, s.SQLAddr(), t.Name(), url.User(username.RootUser))
+		f := makeSinklessFeedFactory(s, sink)
+		return f, cleanup
+	}
+	t.Fatalf("unhandled sink type %s", sinkType)
+	return nil, nil
+}
+
+func cdcTest(t *testing.T, testFn cdcTestFn, testOpts ...feedTestOption) {
+	cdcTestDesc(t, "", testFn, testOpts...)
+}
+
+func cdcTestDesc(t *testing.T, desc string, testFn cdcTestFn, testOpts ...feedTestOption) {
```

Comment:
what does Desc standf for?

Reply:
Description

Reply:
Can it be more descriptive? Desc is  meant as "descriptor" in many parts of the code, that's why I was confused.